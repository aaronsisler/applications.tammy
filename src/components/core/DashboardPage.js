import React from 'react';
import PositionsWatchContainer from 'PositionWatch/PositionsWatchContainer';

const DashboardPage = (props) => (
    <div id="dashboard_page">
        <PositionsWatchContainer {...props} />
    </div>
);

export default DashboardPage;
