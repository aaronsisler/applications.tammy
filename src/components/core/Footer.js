import React from 'react';

const Footer = () => (
    <footer>
        <div>Copyright &copy; {new Date().getFullYear()}</div>
    </footer>
);

export default Footer;
