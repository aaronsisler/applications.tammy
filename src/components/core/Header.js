import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { startLogout } from '../../actions/auth';

const Header = (props) => (
    <header>
        <div className="header_content">
            <Link className="header_title" to="/dashboard">
                Dashboard
                </Link>
            <button className="button" onClick={props.startLogout}>Logout</button>
        </div>
    </header>
);

Header.propTypes = {
    startLogout: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header);
