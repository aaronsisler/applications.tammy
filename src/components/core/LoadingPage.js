import React from 'react';

const LoadingPage = () => (
    <div className="loading_content">
        <img className="loading_image" src="/images/loading.gif" />
    </div>
)

export default LoadingPage;
