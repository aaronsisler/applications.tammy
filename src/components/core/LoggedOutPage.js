import React from 'react';
import { Link } from 'react-router-dom';

const LoggedOutPage = () => (
    <div id="logged_out_page">
        <div className="logged_out_content">
            You have been logged out.
        </div>
        <div className="logged_out_link">
            <Link className="navbar_link" to="/login">Back to Login</Link>
            <Link className="navbar_link" to="/">Back to Positions</Link>
        </div>
    </div>
);

export default LoggedOutPage;
