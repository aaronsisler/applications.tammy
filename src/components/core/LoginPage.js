import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { startLogin } from '../../actions/auth';

const LoginPage = (props) => (
    <div id="login_page">
        <div className="login_content">
            <button className="button" onClick={props.startLogin}>Login with Google</button>
        </div>
    </div>
);

LoginPage.propTypes = {
    startLogin: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
    startLogin: () => dispatch(startLogin('/user_account'))
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
