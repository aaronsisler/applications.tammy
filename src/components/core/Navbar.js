import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { startLogout } from '../../actions/auth';

const Navbar = (props) => (
    <div id="navbar">
        {props.isAuthenticated &&
            <div className="navbar_content">
                <div className="navbar_links">
                    <Link className="navbar_link" to="/dashboard">Dashboard</Link>
                    <Link className="navbar_link" to="/user_account">User Account</Link>
                    <Link className="navbar_link" to="/">Positions</Link>
                </div>
                <div className="navbar_logout">
                    <button className="button navbar_link" onClick={props.startLogout}>Logout</button>
                </div>
            </div>
        }
        {!props.isAuthenticated &&
            <div className="navbar_content">
                <div className="navbar_login">
                    <Link className="navbar_link" to="/login">Login</Link>
                </div>
            </div>
        }
    </div>
);

Navbar.propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    startLogout: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Navbar);
