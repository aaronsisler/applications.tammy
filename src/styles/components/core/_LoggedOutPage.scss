//Small screens - MOBILE
//max-width 640px, small screens
@media only screen {
    .logged_out_content {
        display: flex;
        font-size: $midlarge;
        justify-content: center;
        padding: $large;
    }
    .logged_out_link {
        display: flex;
        justify-content: center;
        padding: 0 $medium $large;
    }
    .navbar_link {
        background: $mainColor;
        color: white;
        font-size: $medium;
        margin: $small;
        padding: $small;
        text-decoration: none;
    }
}

//Medium screens - TABLET
//min-width 641px, medium screens
@media only screen and (min-width: $mediumScreen) {}

//Large screens - DESKTOP
//min-width 1025px, large screens
@media only screen and (min-width: $largeScreen) {
    .logged_out_content {
        justify-content: flex-start;
    }
    .logged_out_link {
        justify-content: flex-start;
    }
}

//XLarge screens
//min-width 1441px, xlarge screens
@media only screen and (min-width: $xlargeScreen) {}

//XXLarge screens
//min-width 1921px, xlarge screens
@media only screen and (min-width: $xxlargeScreen) {}
