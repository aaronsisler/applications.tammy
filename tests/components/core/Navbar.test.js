import React from 'react';
import { shallow } from 'enzyme';
import { Navbar } from '../../../src/components/core/Navbar';

describe('Navbar', () => {
    const startClearUser = jest.fn();
    const startLogout = jest.fn();
    const bodyOpenClassName = 'body_open';
    const navSideMenuOpenClassName = 'nav_side_menu_open';
    const navSideMenu = 'nav_side_menu';
    let wrapper;
    let instance;

    beforeEach(() => {
        // console.log('beforeEach main');
        wrapper = shallow(
            <Navbar
                isAuthenticated={true}
                startClearUser={startClearUser}
                startLogout={startLogout}
            />);
        instance = wrapper.instance();
    })

    afterEach(() => {
        // console.log('Restore mocks');
        jest.restoreAllMocks();
    });

    it('should have isNavOpen set to false on initialization', () => {
        expect(wrapper.state('isNavOpen')).toBe(false);
    });

    describe('favicons', () => {
        beforeEach(() => {
            // console.log('beforeEach spyOn');
            jest.spyOn(instance, 'handleOpenSideMenu').mockImplementation(jest.fn());
            jest.spyOn(instance, 'handleCloseSideMenu').mockImplementation(jest.fn());
        })

        describe('times icon', () => {
            it('should show if isNavOpen is TRUE', () => {
                wrapper.setState({ isNavOpen: true });
                expect(wrapper.state('isNavOpen')).toBe(true);
                expect(wrapper).toMatchSnapshot();
            });

            it('should call handleCloseSideMenu when clicked', () => {
                wrapper.setState({ isNavOpen: true });
                expect(wrapper.state('isNavOpen')).toBe(true);
                wrapper.find('.navbar_favicon').simulate('click');
                expect(instance.handleCloseSideMenu).toHaveBeenCalled();
            });
        });

        describe('bars icon', () => {
            it('should show if isNavOpen is FALSE', () => {
                expect(wrapper.state('isNavOpen')).toBe(false);
                expect(wrapper).toMatchSnapshot();
            });

            xit('should call handleOpenSideMenu when clicked', () => {
                console.log('In test');
                expect(wrapper.state('isNavOpen')).toBe(false);
                // wrapper.find('.navbar_favicon').simulate('click');
                expect(wrapper.find('.navbar_favicon')).toHaveLength(1);
                console.log('After first expect');
                console.log('Open', instance.handleOpenSideMenu);
                console.log('Close', instance.handleCloseSideMenu);
                wrapper.find('#bars_favicon').simulate('click');
                // expect(instance.handleOpenSideMenu).toHaveBeenCalled();
                console.log('End of test');
            });
        });
    });

    describe('handleOpenSideMenu() method', () => {
        let bodyAddMock;
        let getElementByIdAddMock;
        let getElementByIdMock;

        beforeEach(() => {
            instance = wrapper.instance();
            bodyAddMock = jest.fn();
            getElementByIdAddMock = jest.fn();
            const getElementByIdClassListMock = { classList: { add: getElementByIdAddMock } }

            jest.spyOn(document.body.classList, 'add').mockImplementation(bodyAddMock);
            getElementByIdMock = jest.spyOn(document, 'getElementById').mockReturnValue(getElementByIdClassListMock);
        });

        it('should add classname to body classlist', () => {
            instance.handleOpenSideMenu();
            expect(bodyAddMock).toHaveBeenLastCalledWith(bodyOpenClassName);
        });

        it(`should add classname to nav side menu element's classlist`, () => {
            instance.handleOpenSideMenu();
            expect(getElementByIdMock).toHaveBeenLastCalledWith(navSideMenu);
            expect(getElementByIdAddMock).toHaveBeenLastCalledWith(navSideMenuOpenClassName);
        });

        it('should set isNavOpen in state to true', () => {
            instance.handleOpenSideMenu();
            expect(wrapper.state('isNavOpen')).toBe(true);
        });
    })

    describe('handleCloseSideMenu() method', () => {
        let bodyRemoveMock;
        let getElementByIdRemoveMock;
        let getElementByIdMock;

        beforeEach(() => {
            bodyRemoveMock = jest.fn();
            getElementByIdRemoveMock = jest.fn();
            const getElementByIdClassListMock = { classList: { remove: getElementByIdRemoveMock } }

            jest.spyOn(document.body.classList, 'remove').mockImplementation(bodyRemoveMock);
            getElementByIdMock = jest.spyOn(document, 'getElementById').mockReturnValue(getElementByIdClassListMock);
        });

        it('should remove classname to body classlist', () => {
            instance.handleCloseSideMenu();
            expect(bodyRemoveMock).toHaveBeenLastCalledWith(bodyOpenClassName);
        });

        it(`should remove classname to nav side menu element's classlist`, () => {
            instance.handleCloseSideMenu();
            expect(getElementByIdMock).toHaveBeenLastCalledWith(navSideMenu);
            expect(getElementByIdRemoveMock).toHaveBeenLastCalledWith(navSideMenuOpenClassName);
        });

        it('should set isNavOpen in state to false', () => {
            expect(wrapper.state('isNavOpen')).toBe(false);
            wrapper.setState({ isNavOpen: true });
            expect(wrapper.state('isNavOpen')).toBe(true);
            instance.handleCloseSideMenu();
            expect(wrapper.state('isNavOpen')).toBe(false);
        });
    })

    describe('when user is authenticated', () => {
        it('should render Navbar correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });

        describe('handleLogout() method', () => {
            beforeEach(() => {
                jest.spyOn(instance, 'handleOpenSideMenu').mockImplementation(jest.fn());
                jest.spyOn(instance, 'handleCloseSideMenu').mockImplementation(jest.fn());
            })

            it('should call handleCloseSideMenu', () => {
                wrapper.find('button').simulate('click');

                expect(instance.handleCloseSideMenu).toHaveBeenCalled();
            })

            it('should call startClearUser', () => {
                wrapper.find('button').simulate('click');

                expect(startClearUser).toHaveBeenCalled();
            })

            it('should call startLogout', () => {
                wrapper.find('button').simulate('click');

                expect(startLogout).toHaveBeenCalled();
            })
        })
    })

    describe('when user is NOT authenticated', () => {
        beforeEach(() => {
            wrapper = shallow(
                <Navbar
                    isAuthenticated={false}
                    startClearUser={startClearUser}
                    startLogout={startLogout}
                />)
        });

        it('should render Navbar correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    })
})
