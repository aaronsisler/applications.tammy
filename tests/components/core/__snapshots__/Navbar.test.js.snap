// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar favicons bars icon should show if isNavOpen is FALSE 1`] = `
<div
  id="navbar"
>
  <div
    className="nav_toggles"
  >
    <FontAwesomeIcon$1
      border={false}
      className="navbar_favicon"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f0c9",
            "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
          ],
          "iconName": "bars",
          "prefix": "fas",
        }
      }
      listItem={false}
      mask={null}
      name=""
      onClick={[Function]}
      pull={null}
      pulse={false}
      rotation={null}
      size="2x"
      spin={false}
      symbol={false}
      transform={null}
    />
  </div>
  <div
    id="nav_side_menu"
  >
    <Link
      className="nav_side_menu_link"
      onClick={[Function]}
      replace={false}
      to="/dashboard"
    >
      Dashboard
    </Link>
    <Link
      className="nav_side_menu_link"
      onClick={[Function]}
      replace={false}
      to="/user_profile"
    >
      User Profile
    </Link>
    <Link
      className="nav_side_menu_link"
      onClick={[Function]}
      replace={false}
      to="/"
    >
      Positions
    </Link>
    <button
      className="button nav_side_menu_link"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</div>
`;

exports[`Navbar favicons times icon should show if isNavOpen is TRUE 1`] = `
<div
  id="navbar"
>
  <div
    className="nav_toggles"
  >
    <FontAwesomeIcon$1
      border={false}
      className="navbar_favicon"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            352,
            512,
            Array [],
            "f00d",
            "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
          ],
          "iconName": "times",
          "prefix": "fas",
        }
      }
      listItem={false}
      mask={null}
      name=""
      onClick={[MockFunction]}
      pull={null}
      pulse={false}
      rotation={null}
      size="2x"
      spin={false}
      symbol={false}
      transform={null}
    />
  </div>
  <div
    id="nav_side_menu"
  >
    <Link
      className="nav_side_menu_link"
      onClick={[MockFunction]}
      replace={false}
      to="/dashboard"
    >
      Dashboard
    </Link>
    <Link
      className="nav_side_menu_link"
      onClick={[MockFunction]}
      replace={false}
      to="/user_profile"
    >
      User Profile
    </Link>
    <Link
      className="nav_side_menu_link"
      onClick={[MockFunction]}
      replace={false}
      to="/"
    >
      Positions
    </Link>
    <button
      className="button nav_side_menu_link"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</div>
`;

exports[`Navbar when user is NOT authenticated should render Navbar correctly 1`] = `
<div
  id="navbar"
>
  <div
    className="nav_toggles"
  >
    <FontAwesomeIcon$1
      border={false}
      className="navbar_favicon"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f0c9",
            "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
          ],
          "iconName": "bars",
          "prefix": "fas",
        }
      }
      listItem={false}
      mask={null}
      name=""
      onClick={[Function]}
      pull={null}
      pulse={false}
      rotation={null}
      size="2x"
      spin={false}
      symbol={false}
      transform={null}
    />
  </div>
  <div
    id="nav_side_menu"
  >
    <Link
      className="nav_side_menu_link"
      onClick={[Function]}
      replace={false}
      to="/"
    >
      Positions
    </Link>
    <Link
      className="nav_side_menu_link"
      onClick={[Function]}
      replace={false}
      to="/login"
    >
      Login
    </Link>
  </div>
</div>
`;

exports[`Navbar when user is authenticated should render Navbar correctly 1`] = `
<div
  id="navbar"
>
  <div
    className="nav_toggles"
  >
    <FontAwesomeIcon$1
      border={false}
      className="navbar_favicon"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f0c9",
            "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
          ],
          "iconName": "bars",
          "prefix": "fas",
        }
      }
      listItem={false}
      mask={null}
      name=""
      onClick={[Function]}
      pull={null}
      pulse={false}
      rotation={null}
      size="2x"
      spin={false}
      symbol={false}
      transform={null}
    />
  </div>
  <div
    id="nav_side_menu"
  >
    <Link
      className="nav_side_menu_link"
      onClick={[Function]}
      replace={false}
      to="/dashboard"
    >
      Dashboard
    </Link>
    <Link
      className="nav_side_menu_link"
      onClick={[Function]}
      replace={false}
      to="/user_profile"
    >
      User Profile
    </Link>
    <Link
      className="nav_side_menu_link"
      onClick={[Function]}
      replace={false}
      to="/"
    >
      Positions
    </Link>
    <button
      className="button nav_side_menu_link"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</div>
`;
